files/api/user/services/passport.js

```diff
@@ -167,31 +167,22 @@ passport.callback = function * (ctx, next) {
   const provider = params.provider || 'local';
   const action = params.action || 'connect';
 
-  if (provider === 'local' && action !== undefined) {
-    if (action === 'register') {
-      try {
-        let user = yield passport.protocols.local.register(ctx);
-        next(null, user);
-      } catch (err) {
-        next(err);
-      }
-    } else if (action === 'connect') {
-      try {
-        let user = yield passport.protocols.local.login(ctx);
-        next(null, user);
-      } catch (err) {
-        next(err);
-      }
-    } else {
-      next(new Error('Invalid action'));
+  if (provider === 'local' && action === 'register') {
+    try {
+      let user = yield passport.protocols.local.register(ctx);
+      next(null, user);
+    } catch (err) {
+      next(err);
     }
-  } else {
+  } else if (action === 'connect') {
     yield passport.authenticate(ctx.params.provider, function * (err, user) {
```

Comment:
The koa-passport authenticate callback has to be a generator function.

See https://github.com/rkusa/koa-passport/blob/master/lib/framework/koa.js#L78